### ISSUE TITLE
학습이 진행될수록 Loss편차가 커지다가 가중치가 nan이 되버리는 현상

### ISSUE THREAD

**Issue by wolfrev0** [0xC0DEF/A2C#2]
loss의 편차가 너무 크고, loss값 자체도 상당히 큰 음수값이 되서 수치적 문제가 생기는듯.
임시방편으로 Adam의 eps값 추가와 batch size를 128로 늘려서 학습은 꽤나 되는데,
학습되는거와 별개로 몇시간 이상 학습시킬 수가 없이 바로 터져버린다.

유력한 원인들(과 아닌거같은 이유 있으면 화살표로 추가했음)
1. loss가 nan이라서 가중치로 퍼졌다 -> loss가 nan이 되지 않는걸 확인함
2. Adam이 수렴지점 근처에서 매우 큰 학습률을 가져서 터진다 -> 아닌거같음
3. 학습이 너무 많이되서 과적합된거다 -> 성능이 하락하는건 아니고 유지되거나 개선되다가 터짐
4. loss함수의 결함

 ~~현재로써는 일단 loss함수의 결함이라고 생각되는데~~,
https://www.google.com/search?q=a2c+negative+loss&newwindow=1&sxsrf=ALeKk02Xe_TyYTnM8iFyFA_VyNt4rzKjXA%3A1617055001770&source=hp&ei=GU1iYKvpLIGVr7wP2KqWmAQ&iflsig=AINFCbYAAAAAYGJbKWYPpEeTkQDSo56jKtWJvlOfS05L&oq=a2c+negat&gs_lcp=Cgdnd3Mtd2l6EAMYADIGCAAQFhAeOgQIIxAnOgUIABCRAjoCCAA6CAguEMcBEKMCOgQIABBDOgQIABAKOggIABAWEAoQHlDSAVidF2CrHWgAcAB4AIABe4gBigiSAQMwLjmYAQCgAQGqAQdnd3Mtd2l6&sclient=gws-wiz
https://ai.stackexchange.com/questions/2405/how-do-i-handle-negative-rewards-in-policy-gradients-with-the-cross-entropy-loss
~~위 링크 자료들 살펴보면 cross entropy loss가 reward가 양수임을 조건으로 한다는듯.~~
~~그래서 음수쪽 학습이 잘못되서 loss가 점점 큰 음수가 되는게 아닐까 일단 그렇게 생각된다.~~

**Comment by wolfrev0** [comment]
Batch Normalization을 전부다 제거하니까 해결됨. 배치크기가 너무 작아서 그런가? 일단 학습이 잘 되니까 정규화 및 학습가속 의미를 가지는 배치노말은 없어도 되겠다.
